hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -332,10 +356,40 @@ def from_typing_type(thing):
             # Reversible is somehow a subclass of Hashable, so we tuplize it.
             # See also the discussion at https://bugs.python.org/issue39046
             typing.Reversible: st.lists(st.integers()).map(tuple),  # type: ignore
-            typing.SupportsAbs: st.complex_numbers(),
-            typing.SupportsComplex: st.complex_numbers(),
-            typing.SupportsFloat: st.floats(),
-            typing.SupportsInt: st.floats(),
+            typing.SupportsAbs: st.one_of([st.booleans(),
```

Comment:
For these statements, you'll need to run `black .` to get the formatting right.

I'd also suggest using the multi-arg form of `one_of`, to avoid a layer of indentation due to the list - `one_of(st.booleans(), st.integers(), ...)`

Reply:
ah yes multiarg is nicer!. 
ah we use black.... that will save me a lot of faff.